lib/matplotlib/patches.py

```diff
@@ -3223,12 +3223,10 @@ def __call__(self, path, mutation_size, linewidth,
             and takes care of the aspect ratio.
             """
 
-            path = make_path_regular(path)
-
             if aspect_ratio is not None:
                 # Squeeze the given height by the aspect_ratio
 
-                vertices, codes = path.vertices[:], path.codes[:]
+                vertices, codes = path.vertices[:], path.codes
```

Comment:
No need to create a variable for the `codes`:

~~~
vertices = path.vertices[:]
# Squeeze the height
vertices[:, 1] = path.vertices[:, 1] / aspect_ratio
path_shrunk = Path(vertices, path.codes)
~~~

Reply:
replaced the whole thing to use broadcasting, which is shorter.